From 671dca5f39e22734a789f8a33a563aeae790ef3b Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 31 Jul 2013 11:07:24 +0800
Subject: [PATCH 14/23] ENGR00273268 add YUV format support in libagl
 eglCreateImageKHR

add YUV format support in libagl eglCreateImageKHR,
else libgui surfacetexture createImage report fail
on mx6sl platform when play video.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 opengl/libagl/egl.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/opengl/libagl/egl.cpp b/opengl/libagl/egl.cpp
index 2035cee..d23c26a 100644
--- a/opengl/libagl/egl.cpp
+++ b/opengl/libagl/egl.cpp
@@ -2141,6 +2141,16 @@ EGLImageKHR eglCreateImageKHR(EGLDisplay dpy, EGLContext ctx, EGLenum target,
         case HAL_PIXEL_FORMAT_RGB_565:
         case HAL_PIXEL_FORMAT_BGRA_8888:
             break;
+        //add YUV format support
+        case HAL_PIXEL_FORMAT_YV12:
+        case HAL_PIXEL_FORMAT_YCbCr_422_SP:
+        case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+        case HAL_PIXEL_FORMAT_YCbCr_422_I:
+        case HAL_PIXEL_FORMAT_YCbCr_422_P:
+        case HAL_PIXEL_FORMAT_YCbCr_420_P:
+        case HAL_PIXEL_FORMAT_CbYCrY_422_I:
+        case HAL_PIXEL_FORMAT_YCbCr_420_SP:
+            break;
         default:
             return setError(EGL_BAD_PARAMETER, EGL_NO_IMAGE_KHR);
     }
-- 
1.8.0

